RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplate2D.cc

```diff
@@ -541,19 +541,21 @@ bool SiPixelTemplate2D::interpolate(int id, float cotalpha, float cotbeta, float
    
    // Check angle limits and et up interpolation parameters
    
-   if(cotalpha < cotalpha0_) {
+   acota = std::abs(cotalpha);

```

Comment:
Are ```cotalpha0_``` and ```cotalpha1_``` (as well as ```cotbeta0_``` and ``` cotbeta1_```) assured to be >0? 
That's not straightforward to get from just reading the code...

Reply:
They are not _assured_ to be >0 by checking/catching it in the code, but since they come from the template files that are read in from the database, they are always fixed by us. `cotapha1_` and `cotbeta_1` are therefore definitely >0. For historical reasons we still have some templates around with `cotapha0_<0`, but this will not result in any problems.

Reply:
Shouldn't a check be allowed then in the code, to deal with such "historical" templates or to some possible new "wrong" one?
Why do you say that this wouldn't result in any problem? I don't think that interpolation will be correctly dealt with if the interval spans also over negative values and you use the absolute value for cotalpha/cotbeta...

Reply:
@perrotta I can check but I am pretty sure that all templates *that we can possibly use* will assume cotalpha and cotbeta > 0.    (That is, the templates were this was not true were from 2005 or something.)

Reply:
You already have assert's that check the correctness of what had ended up in the template, e.g.
https://github.com/schuetzepaul/cmssw/blob/7e5e26285e03f3c61f52f7d4c5d63deee92d8ea8/RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplate2D.cc#L515
I think that, insteady of silently giving wrong results, which could escape being noticed, one should rather also
``` 
   assert (cotalpha0_>0 && cotalpha1_>cotalpha0_);
```
and
``` 
   assert (cotbeta0_>0 && cotbeta1_>cotbeta0_);
```

On the other end, it is probably an overkill to check them (but then also ```Nyx_```  and  ```Nxx_```) at every step in the interpolation: couldn't all them be more conveniently checked once for all in the ```pusfhile(...)``` methods?

Reply:
@perrotta 
Yeah, I think doing that in pushfile() is probably a good idea.